core/src/toga/widgets/slider.py

```diff
@@ -1,191 +1,218 @@
-import warnings
+from abc import ABC, abstractmethod
+from contextlib import contextmanager
+from typing import Optional, Tuple
 
 from toga.handlers import wrapped_handler
 
 from .base import Widget
 
 
 class Slider(Widget):
-    """Slider widget, displays a range of values.
-
-    Args:
-        id: An identifier for this widget.
-        style (:obj:`Style`):
-        value (float): Initial value of the slider
-        range (``tuple``): Min and max values of the slider in this form (min, max).
-        tick_count (``int``): How many ticks in range. if None, slider is continuous.
-        on_change (``callable``): The handler to invoke when the slider value changes.
-        on_press (``callable``): The handler to invoke when the slider has been
-            pressed.
-        on_release (``callable``): The handler to invoke when the slider has been
-            released.
-        enabled (bool): Whether user interaction is possible or not.
-    """
-
     def __init__(
         self,
         id=None,
         style=None,
         value=None,
-        range=None,
+        range=(0, 1),
         tick_count=None,
         on_change=None,
-        on_slide=None,  # DEPRECATED!
         on_press=None,
         on_release=None,
         enabled=True,
-        factory=None,  # DEPRECATED!
-        default=None,  # DEPRECATED!
     ):
+        """Create a new slider widget.
+
+        Inherits from :class:`~toga.widgets.base.Widget`.
+
+        :param id: The ID for the widget.
+        :param style: A style object. If no style is provided, a default style
+            will be applied to the widget.
+        :param value: Initial :any:`value`: defaults to the mid-point of the range.
+        :param range: Initial :any:`range`: defaults to (0, 1).
```

Comment:
Should these be semicolons, rather than colons?
```suggestion
        :param value: Initial :any:`value`; defaults to the mid-point of the range.
        :param range: Initial :any:`range`; defaults to (0, 1).
```

Reply:
My understanding is that the semicolon is for when both halves are complete sentences, and the colon for when the second half explains the first half. The first Google result for this question seems to agree:
* https://www.sussex.ac.uk/informatics/punctuation/colonandsemi/semi
* https://www.sussex.ac.uk/informatics/punctuation/colonandsemi/colon

Reply:
Not sure I agree with your analysis here - the definition of semicolon as:
> A semicolon can always, in principle, be replaced either by a [full stop](https://www.sussex.ac.uk/informatics/punctuation/stopsandmarks/full) (yielding two separate sentences) or by the word and (possibly preceded by a [joining comma](https://www.sussex.ac.uk/informatics/punctuation/comma/joining)). 

seems exactly what we need here, whereas:

> The colon is used to indicate that what follows it is an explanation or elaboration of what precedes it. T

Doesn't seem to apply here - I wouldn't call the definition of the default as an explanation or elaboration.

At the very least, splitting into 2 sentences works. And, given that the woman who reviewed my PhD thesis threatened to remove the colon and semicolon keys from my keyboard, I'm inclined to take her advice and just use a full stop :-)